### ISSUE TITLE
RHEL Based Containers

### ISSUE THREAD

**Issue by bsahane** [spurin/diveintoansible-images#1]
Hello @spurin,

I tried to modify the containers and instead of CentOS (`spurin/container-systemd-sshd-ttyd:centos_8`) I am trying to use official RHEL Image (`registry.access.redhat.com/ubi8/ubi:8.5-236.1648460182`) from https://catalog.redhat.com/software/containers/ubi8/ubi/5c359854d70cc534b3a3784e?container-tabs=gti

When ever I tried to run the `docker-compose up` then I am getting error `$container exited with code 0`

Can you please help me to build the Same environment with RHEL.

**Comment by spurin** [comment]
Hi @bsahane 

This sounds like an interesting area, let's give it a go.  

We'd need to go up a few layers though and start in https://github.com/spurin/container-systemd

The ubi image will be barebone so it may need to have systemd etc installed.  The existing centos branches will be the best example to work from.

Have a play around starting there, I'll also try and take a look 👍

**Comment by bsahane** [comment]
Thank You @spurin for your Guidance.
I will verify with https://github.com/spurin/container-systemd repository

**Comment by bsahane** [comment]
Hello @spurin,
Thank You, I update the image with "systemd" as per given in the [above](https://github.com/spurin/container-systemd/tree/centos_8) repository[1].
(update `Dockerfile` and run `sh build.sh --local`)
Now container is not exit but still cant accessible via ssh or webui

Do I need to execute [container-entrypoint.sh](https://github.com/spurin/container-systemd/blob/centos_8/container-entrypoint.sh) or need to do any other steps.
Please guide me as I tried to find for `container-entrypoint.sh` but not found the steps in README.md

**Comment by spurin** [comment]
Hi @bsahane 

Okay great, we’re you able to test it as per the readme in this repo?

The container images are stacked so this is the base, then, there’s another called container-systemd-sshd-ttyd which is also in my repo list, you’d then build that and reference this image.

Then lastly, you’d build a diveintoansible-images:rhel branch, I would try using the centos one as a guide here.

Its a good learning experience so give it a go and let me know how you get on, happy to help with any problems 👍

**Comment by bsahane** [comment]
Thank you @spurin I will verify with `systemd-sshd-ttyd`

**Comment by bsahane** [comment]
Hello @spurin,
I tried Multiple Times with different RHEL Available Container Images,
But I am getting below error after configuring container with [container-systemd](https://github.com/spurin/container-systemd) repository

Command,
```
docker run -it --rm --privileged $container /bin/bash
```
Error Screenshot

![2022-05-01_18-21](https://user-images.githubusercontent.com/3491929/166149404-2158ae7e-3edf-41d8-89b1-83de9d330543.png)

While Your provided containers are working fine as

![image](https://user-images.githubusercontent.com/3491929/166149440-62165826-393d-41e1-becf-88a19b5593ea.png)

**Comment by spurin** [comment]
This may be something g to do with the UBI images possibly being rootless.

I'll try and have a play around and see if I can get it going.

Full transparency, I may be a little while!  Am currently working on what will be a free Kubernetes course and am trying to get this out asap!

**Comment by bsahane** [comment]
Sure Thank You @spurin,
Please take your Time.
Till I will try myself and update if I found any solution.

**Comment by spurin** [comment]
Just thinking about this.  What happens if you run the container without a command?

**Comment by bsahane** [comment]
Yes @spurin, It is working but I can not access it from Web Portal or Using the default ssh ip:port

**Comment by spurin** [comment]
Okay so it's working but not allowing you to login?

**Comment by bsahane** [comment]
Yes Spurin, 
the container is running in background and visible on http://localhost:1000/ but when I tried to login using 

Web Interface then Showing error as

![image](https://user-images.githubusercontent.com/3491929/166704465-40adcf96-9138-4ce5-bfb7-da95b893537e.png)
-------------------------------------------------------------------------------------------------
and for login using SSH showing error as
```bash
ssh: connect to host 172.18.0.1 port $PORT: Connection refused
```

**Comment by spurin** [comment]
Hmm so it looks like the services may not be starting up.  Ping me some copies of the Dockerfiles and I’ll take a look

**Comment by bsahane** [comment]
Sure Spurin,
I will share the files

**Comment by bsahane** [comment]
Hello @spurin,

Please find the below attachment of all modifications I have done in the Source.
https://github.com/bsahane/lab2/raw/master/lab2.zip

I pushed the RHEL8 Container Image with Subscription to DockerHub as : sahanebhushan7/container:rhel8

**Comment by bsahane** [comment]
Here first I updated
1. diveintoansible-images > Dockerfile as `FROM sahanebhushan7/container:rhel8` and `build.sh`
2. container-systemd > Dockerfile as `FROM sahanebhushan7/container:rhel8` (commented Centos8 EOL) and `build.sh`
3. container-systemd-sshd-ttyd > Dockerfile as `FROM sahanebhushan7/container:rhel8 as loginbuild` (commented Centos8 EOL), `FROM sahanebhushan7/container:rhel8` and `build.sh`
4. diveintoansible-lab > `docker-compose.yaml` as For `centos1:` `image: sahanebhushan7/container:rhel8`
